lib/client.js

```diff
@@ -509,8 +535,23 @@ Client.prototype.setupBroker = function (host, port, longpolling, brokers) {
 };
 
 Client.prototype.createBroker = function connect(host, port, longpolling) {
+
+    var addr = host + ':' + port;
+
+    //prefer host and port from broker discovery
+    //this gets populated by the initial discovery of brokers from kafka
+    var brokerProfile = this.brokerProfiles[addr];
+
+    host = brokerProfile[this.ssl ? 'sslHost' : 'host'] || host;
+    port = brokerProfile[this.ssl ? 'sslPort' : 'port'] || port;
+
     var self = this;
-    var socket = net.createConnection(port, host);
+    var socket;
+    if (self.ssl) {
+        socket = tls.connect(port, host, self.sslOptions);
+    } else {
+        socket = net.createConnection(port, host);
+    }
     socket.addr = host + ':' + port;
```

Comment:
`addr` may have changed because of :545/:546 it is important that we recalculate it here.